Sofa/framework/Geometry/src/sofa/geometry/Edge.h

```diff
@@ -121,6 +121,51 @@ struct Edge
     }
 
 
+    /**
+    * @brief	Test if a point is on Edge (n0, n1)
+    * @tparam   Node iterable container
+    * @tparam   T scalar
+    * @param	p0: position of the point to test
+    * @param	n0,n1: nodes of the edge
+    * @return	bool result if point is on Edge.
+    */
+    template<typename Node,
+        typename T = std::decay_t<decltype(*std::begin(std::declval<Node>()))>,
+        typename = std::enable_if_t<std::is_scalar_v<T>>
+    >
+        static constexpr bool isPointOnEdge(const Node& p0, const Node& n0, const Node& n1)
+    {
+        // 1. check if the point, n0 and n1 are aligned
+        const auto AB = n0 - n1;
+        const auto AC = n0 - p0;
+
+        T N2 = T(0);
+        if constexpr (std::is_same_v < Node, sofa::type::Vec<2, T> >)
+        {            
+            N2 = sofa::type::cross(AB, AC);
+        }
+        else
+        {
+            N2 = sofa::type::cross(AB, AC).norm2();
+        }
+
+        if (N2 > EQUALITY_THRESHOLD || N2 < 0.0f)
+            return false;
+
+
+        // 2. check if point is between {n0; n1}
+        const auto Ln0p0 = sofa::type::dot(AB, AC);
+        const auto Ln0n1 = sofa::type::dot(AB, AB);
+
+        if (Ln0p0 < 0.0f || Ln0p0 > Ln0n1) // out of bounds [n0; n1]
+            return false;
+        else if (Ln0n1 < EQUALITY_THRESHOLD) // null edge
```

Comment:
Couldn't this be tested directly at beginning ? (little optim, depending on the cross product implementation it could even avoid numerical issues...)

Reply:
For the rest, I've looked at the geometrical point of view and the function is correct, it return true for a point that is between the two points and at a distance of at  most EQUALITY_THRESHOLD.

Reply:
Done